firmware/user/modes/mode_tunernome.c

```diff
@@ -0,0 +1,721 @@
+            setLeds(leds, sizeof(leds));
+
+            clearDisplay();
+            break;
+        }
+        default:
+        {
+            break;
+        }
+    }
+}
+
+/**
+ * Called when tunernome is exited
+ */
+void ICACHE_FLASH_ATTR tunernomeExitMode(void)
+{
+    timerDisarm(&(tunernome->updateTimer));
+    timerFlush();
+    os_free(tunernome);
+}
+
+/**
+ * Inline helper function to get the magnitude of a frequency bin from fuzzed_bins[]
+ *
+ * @param idx The index to get a magnitude from
+ * @return A signed magnitude, even though fuzzed_bins[] is unsigned
+ */
+static inline int16_t getMagnitude(uint16_t idx)
+{
+    return fuzzed_bins[idx];
+}
+
+/**
+ * Inline helper function to get the difference in magnitudes around a given
+ * frequency bin from fuzzed_bins[]
+ *
+ * @param idx The index to get a difference in magnitudes around
+ * @return The difference in magnitudes of the bins before and after the given index
+ */
+static inline int16_t getDiffAround(uint16_t idx)
+{
+    return getMagnitude(idx + 1) - getMagnitude(idx - 1);
+}
+
+/**
+ * Inline helper function to get the magnitude of a frequency bin from folded_bins[]
+ *
+ * @param idx The index to get a magnitude from
+ * @return A signed magnitude, even though folded_bins[] is unsigned
+ */
+static inline int16_t getSemiMagnitude(int16_t idx)
+{
+    if(idx < 0)
+    {
+        idx += FIXBPERO;
+    }
+    if(idx > FIXBPERO - 1)
+    {
+        idx -= FIXBPERO;
+    }
+    return folded_bins[idx];
+}
+
+/**
+ * Inline helper function to get the difference in magnitudes around a given
+ * frequency bin from folded_bins[]
+ *
+ * @param idx The index to get a difference in magnitudes around
+ * @return The difference in magnitudes of the bins before and after the given index
+ */
+static inline int16_t getSemiDiffAround(uint16_t idx)
+{
+    return getSemiMagnitude(idx + 1) - getSemiMagnitude(idx - 1);
+}
+
+/**
+ * Recalculate the per-bpm values for the metronome
+ */
+void ICACHE_FLASH_ATTR recalcMetronome() {
+    tunernome->bps = tunernome->bpm / 60.0f;
+    tunernome->periodS = 1.0f / tunernome->bps;
+    tunernome->periodMS = 1000.0f * tunernome->periodS;
+    tunernome->cycleFrames = tunernome->periodMS / TUNERNOME_UPDATE_MS;
+    tunernome->halfCycleFrames = tunernome->cycleFrames * 0.5f;
+    tunernome->tockFrame1 = round(tunernome->cycleFrames * 0.25f);
+    tunernome->tockFrame2 = round(tunernome->cycleFrames * 0.75f);
+    tunernome->finalCycleFrame = round(tunernome->cycleFrames);
+}
+
+/**
+ * @brief called on a timer, updates the game state
+ *
+ * @param arg
+ */
+static void ICACHE_FLASH_ATTR tunernomeUpdate(void* arg __attribute__((unused)))
+{
+    switch(tunernome->mode)
+    {
+        default:
+        case TN_TUNER:
+        {
+            plotText(0, OLED_HEIGHT - FONT_HEIGHT_TOMTHUMB, "< Exit", TOM_THUMB, WHITE);
+            plotText(OLED_WIDTH - 43, OLED_HEIGHT - FONT_HEIGHT_TOMTHUMB, "Metronome >", TOM_THUMB, WHITE);
```

Comment:
You should use `textWidth()` to measure the text for right justification, rather than the hardcoded `43`.

Reply:
I just discovered that function earlier today, so I'll switch to it here too.